File: database/models/alert.py
import uuid

from sqlalchemy import Boolean, Column, DateTime, Numeric, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func

from database.base import Base


class Alert(Base):
    __tablename__ = "alerts"

    id = Column(
        UUID(as_uuid=True),
        primary_key=True,
        nullable=False,
        default=uuid.uuid4,
    )

    client_id = Column("clientid", UUID(as_uuid=True), nullable=False)
    installation_id = Column("installationid", UUID(as_uuid=True), nullable=True)
    plant_id = Column("plantid", UUID(as_uuid=True), nullable=True)
    reading_type_id = Column("readingtypeid", UUID(as_uuid=True), nullable=False)

    name = Column(String(150), nullable=False)
    description = Column(String(500), nullable=True)

    channel = Column(String(30), nullable=False)
    recipient_email = Column("recipientemail", String(255), nullable=True)

    condition_type = Column("conditiontype", String(30), nullable=False)
    value_type = Column("valuetype", String(20), nullable=False)

    min_value = Column("minvalue", Numeric(18, 4), nullable=True)
    max_value = Column("maxvalue", Numeric(18, 4), nullable=True)
    exact_numeric_value = Column("exactnumericvalue", Numeric(18, 4), nullable=True)
    exact_text_value = Column("exacttextvalue", String(255), nullable=True)
    exact_boolean_value = Column("exactbooleanvalue", Boolean, nullable=True)

    is_active = Column("isactive", Boolean, nullable=False, default=True)
    is_deleted = Column("isdeleted", Boolean, nullable=False, default=False)

    created_on = Column("createdon", DateTime, nullable=False, server_default=func.now())
    created_by = Column("createdby", String(100), nullable=True)
    modified_on = Column("modifiedon", DateTime, nullable=True)
    modified_by = Column("modifiedby", String(100), nullable=True)
    deleted_on = Column("deletedon", DateTime, nullable=True)

File: backend/greenlytics-api-v3-dotnet/src/GreenLytics.V3.Domain/Entities/Alert.cs
using GreenLytics.V3.Domain.Common;

namespace GreenLytics.V3.Domain.Entities;

public sealed class Alert : BaseAuditableEntity
{
    public Guid ClientId { get; set; }
    public Guid? InstallationId { get; set; }
    public Guid? PlantId { get; set; }
    public Guid ReadingTypeId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string Channel { get; set; } = string.Empty;
    public string? RecipientEmail { get; set; }
    public string ConditionType { get; set; } = string.Empty;
    public string ValueType { get; set; } = string.Empty;
    public decimal? MinValue { get; set; }
    public decimal? MaxValue { get; set; }
    public decimal? ExactNumericValue { get; set; }
    public string? ExactTextValue { get; set; }
    public bool? ExactBooleanValue { get; set; }
    public bool IsActive { get; set; } = true;
}
